### DISCUSSION TITLE
Get query string before executing

### DISCUSSION THREAD

**Discussion by enriquinho-catalan** [byme8/ZeroQL#50] [Ideas]
There are some graphQL queries that consume another query in the form of a string such as Shopify's bulk operations:
https://shopify.dev/docs/api/usage/bulk-operations/queries

Would it be possible to generate queries and serialize them to a string before sending a request to acommodate this use case?

**Comment by byme8** [comment]
At the moment, there is no easy way to do it. I will have a look at how to support it.

**Reply by byme8** [reply]
@enriquinho-catalan, checkout v4.2.0-preview.1. Now it is possible to get raw graphql from lambda:
``` csharp
var queryInfo = QueryInfoProvider.Materialize<Query>(m => m.Me(o => o.FirstName));
// {
//    Query: query { me { firstName } },
//    QueryBody: { me { firstName } },
//    OperationType: query,
//    Hash: 8ed4d3e773b6f87d986cc128a716cfc85d030c5fe6a5b585ab0c0820ac5d9728
// }
```
